### PR TITLE
fix(TCOMP-696): Protect from SQL injection in TCOMP/JDBC component

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TCOMP-696

**What is the new behavior?**
we will validate the sql if a query

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: components/components-jdbc/components-jdbc-definition/src/test/java/org/talend/components/jdbc/dataset/JDBCDatasetPropertiesTest.java [hunk 1]
```diff
@@ -74,12 +75,45 @@ public void testTrigger() {
...
+    
+    @Test(expected = TalendRuntimeException.class)
+    public void testGetSqlWithQueryCheck3() {
+        dataset.sql.setValue("select * \n from abc for update  ");
+        dataset.sourceType.setValue(SourceType.QUERY);
+        dataset.getSql();
+    }
+    
+    @Test(expected = TalendRuntimeException.class)
+    public void testGetSqlWithQueryCheck4() {
```
Comment (pteyssier) [inline components/components-jdbc/components-jdbc-definition/src/test/java/org/talend/components/jdbc/dataset/JDBCDatasetPropertiesTest.java]: Either a short comment or method name more explicit would help in the future to understand what is the test about. Avoid "Check1..2..3" in method name.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1476/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1476/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTCOMP-696)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1477/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1477/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTCOMP-696)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1478/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1478/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTCOMP-696)

File: components/components-jdbc/components-jdbc-definition/src/main/java/org/talend/components/jdbc/CommonUtils.java [hunk 2]
```diff
@@ -172,4 +174,24 @@ public static void setCommonConnectionInfo(AllSetting setting, JDBCConnectionMod
         setting.setPassword(connection.userPassword.password.getValue());
     }
 
+    private static Pattern pattern = Pattern.compile("^SELECT\\s+((?!((INTO)|(FOR\\s+UPDATE)|(LOCK\\s+IN\\s+SHARE\\s+MODE))).)+$",
+            Pattern.CASE_INSENSITIVE | Pattern.DOTALL | Pattern.MULTILINE);
+
+    /**
+     * validate if the sql is a pure query statement which don't write or lock something, if a query, return it, if not, throw
+     * some
```
Comment (jzhao-talend) [inline components/components-jdbc/components-jdbc-definition/src/main/java/org/talend/components/jdbc/CommonUtils.java]: comment format issue.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1479/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1479/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awwang-talend%2FTCOMP-696)

**Pull request merged**